src/ray/core_worker/actor_handle.cc

```diff
@@ -95,4 +95,14 @@ void ActorHandle::SetActorTaskSpec(TaskSpecBuilder &builder, const ObjectID new_
 
 void ActorHandle::Serialize(std::string *output) { inner_.SerializeToString(output); }
 
+bool ActorHandle::IsPersistedToGCS() const {
+  absl::MutexLock guard(&mutex_);
+  return is_persisted_to_gcs_;
+}
+
+void ActorHandle::SetIsPersistedToGCSFlag() {
+  absl::MutexLock guard(&mutex_);
+  is_persisted_to_gcs_ = true;
```

Comment:
I think having this will make the logic cleaner. If we don't have this, we should make sure the actor id is inside `actors_pending_location_resolution_ ` every time IO call happens (to avoid race condition).

Reply:
I decided to go with your suggestion. I will remove actor handle internal state. It requires some amount of refactoring, so please review it one more time!